aten/src/ATen/native/SpectralOps.cpp

```diff
@@ -993,4 +994,93 @@ Tensor istft(const Tensor& self, const int64_t n_fft, const optional<int64_t> ho
       onesidedOpt, lengthOpt, /*return_complex=*/false);
 }
 
+void _fft_fill_with_conjugate_symmetry_(const Tensor& input, IntArrayRef dim_) {
+  const auto input_sizes = input.sizes();
+  const auto input_strides = input.strides();
+  TORCH_CHECK(dim_.size() > 0);
+  DimVector dim(dim_.begin(), dim_.end());
+  at::maybe_wrap_dims(dim, input_strides.size());
+
+  if (input.numel() == 0 || input_sizes[dim.back()] <= 2) {
+    return;  // No elements need writing
+  }
+
+  // Small dimensions may be treated as batch dims since they don't get mirrored
+  dim.erase(
+      std::remove_if(dim.begin(), dim.end(), [&](int64_t dim) {
+        return (input_sizes[dim] <= 2);
+      }),
+      dim.end());
+
+  // Use TensorIterator to coalesce batch dimensions
+  // NOTE: Can't use TensorIterator loops because we need negative strides
+  auto iter = TensorIteratorConfig()
+      .add_output(input)
+      .add_input(input)
+      .resize_outputs(false)
+      .declare_static_shape(input_sizes, dim)
+      .build();
+
+  const auto iter_strides = iter.strides(0);
+  const auto iter_sizes = iter.shape();
+  const auto ndim = iter_strides.size() + dim.size();
+  DimVector in_strides(ndim), signal_half_sizes(ndim);
+
+  const auto element_size = iter.element_size(0);
+  auto* data_ptr = static_cast<char*>(input.data_ptr());
+  for (int64_t i = 0; i < iter_strides.size(); ++i) {
+    in_strides[i] = iter_strides[i];
+    signal_half_sizes[i] = iter_sizes[i];
+  }
+  for (int64_t i = 0; i < dim.size(); ++i) {
+    // Convert to byte strides to match TensorIterator
+    in_strides[iter_strides.size() + i] = input_strides[dim[i]] * element_size;
+    signal_half_sizes[iter_strides.size() + i] = input_sizes[dim[i]];
+  }
+  signal_half_sizes.back() = (input_sizes[dim.back()] - 1) / 2;
+  auto out_strides = in_strides;
+  out_strides.back() *= -1;
+
+  const auto* in_data = data_ptr + input_strides[dim.back()] * element_size;
+  auto* out_data = data_ptr + (
+      input_strides[dim.back()] * (input_sizes[dim.back()] - 1) * element_size);
+
+  // Reorder dimensions by stride to maximize data locality
+  DimVector dim_permute(ndim);
+  std::iota(dim_permute.begin(), dim_permute.end(), 0);
+  std::sort(dim_permute.begin(), dim_permute.end(),
+      [&](auto dim1, auto dim2) {
+        return in_strides[dim1] < in_strides[dim2];
+      });
+
+  DimVector temp(ndim);
+  for (int64_t i = 0; i < ndim; ++i) {
```

Comment:
Add a comment explaining this bit. The sleight-of-hand with temp, in_strides, and out_strides is not obvious

Reply:
There must be a better way to do this vs. copying the code 3 times, right?

Reply:
I've moved it out into a lambda which I call 3 times.